WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductFromImage/AddProductFromImageViewModel.swift

```diff
@@ -147,18 +156,34 @@ private extension AddProductFromImageViewModel {
             do {
                 let texts = try await imageTextScanner.scanText(from: image)
                 analytics.track(event: .AddProductFromImage.scanCompleted(source: addProductSource, scannedTextCount: texts.count))
+
+                guard texts.isNotEmpty else {
+                    throw ScanError.noTextDetected
+                }
+
                 scannedTexts = texts.map { .init(text: $0, isSelected: true) }
                 [nameViewModel, descriptionViewModel].forEach { $0.reset() }
                 generateProductDetails()
             } catch {
-                analytics.track(event: .AddProductFromImage.scanFailed(source: addProductSource, error: error))
-                DDLogError("⛔️ Error scanning text from image: \(error)")
+                switch error {
+                case ScanError.noTextDetected:
+                    if scannedTexts.isEmpty {
+                        textDetectionErrorMessage = Localization.noTextDetected
+                    }
+                    DDLogError("⛔️ No text detected from image.")
+                default:
+                    analytics.track(event: .AddProductFromImage.scanFailed(source: addProductSource, error: error))
+                    if scannedTexts.isEmpty {
```

Comment:
I think this check redundant 🤔

Reply:
https://github.com/woocommerce/woocommerce-ios/pull/10282#discussion_r1270497654

In the above comment, I explained why we have this check. Kindly let me know if I am missing something.